src/models/Execution.ts

```diff
@@ -51,7 +51,7 @@ export default class Execution extends EventEmitter {
 
     onStateUpdate(state, event) {
         this.emit('update', state, event);
-        if (event.failureType && event.failedCommands) {
+        if (event && event.failureType && event.failedCommands) {
```

Comment:
```suggestion
        if (event?.failureType && event?.failedCommands) {
```

You could also leverage [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html), which is supported since Typescript 3.7.

Not sure what codestyles are used here :-).